bin-src/tasks/gitInfo.ts

```diff
@@ -150,44 +146,54 @@ export const setGitInfo = async (ctx: Context, task: Task) => {
     }
 
     const baselineBuilds = await getBaselineBuilds(ctx, { branch, parentCommits });
-    const baselineCommits = baselineBuilds.map((build) => build.commit);
-    ctx.log.debug(`Found baselineCommits: ${baselineCommits.join(', ')}`);
+    ctx.git.baselineCommits = baselineBuilds.map((build) => build.commit);
+    ctx.log.debug(`Found baselineCommits: ${ctx.git.baselineCommits.join(', ')}`);
 
     // Use the most recent baseline to determine final CLI output if we end up skipping the build.
     // Note this will get overwritten if we end up not skipping the build.
     ctx.build = baselineBuilds.sort((a, b) => b.committedAt - a.committedAt)[0] as any;
 
     try {
-      const results = await Promise.all(
-        baselineBuilds.map(async (build) => ({
-          build,
-          ...(await getChangedFilesWithReplacement(ctx, build)),
-        }))
+      // Map the baseline builds to their changed files, falling back to an earlier "replacement"
+      // build if the baseline commit no longer exists (e.g. due to force-pushing).
+      const changedFilesWithInfo = await Promise.all(
+        baselineBuilds.map(async (build) => {
+          const changedFilesWithReplacement = await getChangedFilesWithReplacement(ctx, build);
+          return { build, ...changedFilesWithReplacement };
+        })
+      );
+
+      // Take the distinct union of changed files across all baselines.
+      ctx.git.changedFiles = Array.from(
+        new Set(changedFilesWithInfo.flatMap(({ changedFiles }) => changedFiles))
       );
-      ctx.git.changedFiles = Array.from(new Set(results.flatMap((r) => r.changedFiles)));
 
-      ctx.git.replacementBuildIds = results
+      // Track changed package manifest files along with the commit they were changed in.
+      ctx.git.packageManifestChanges = changedFilesWithInfo.flatMap(({ build, changedFiles }) => {
+        const manifestFiles = changedFiles.filter(isPackageManifestFile);
+        return manifestFiles.length ? [{ changedFiles: manifestFiles, commit: build.commit }] : [];
```

Comment:
I think you'd want to use the `replacementBuild.commit` if it was set over the `build.commit`.